master/internal/api_generic_tasks.go

```diff
@@ -433,12 +458,162 @@ func (a *apiServer) KillGenericTask(
 	return &apiv1.KillGenericTaskResponse{}, nil
 }
 
+func (a *apiServer) PauseGenericTask(
+	ctx context.Context, req *apiv1.PauseGenericTaskRequest,
+) (*apiv1.PauseGenericTaskResponse, error) {
+	var taskModel model.Task
+	err := db.Bun().NewSelect().Model(&taskModel).Where("task_id = ?", req.TaskId).Scan(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// Check if the task is in a state which allows pausing.
+	overrideStates := []model.TaskState{model.TaskStateCanceled, model.TaskStateCompleted, model.TaskStatePaused}
+	// Validate state
+	if slices.Contains(overrideStates, *taskModel.State) {
```

Comment:
when running this against a non-generic task, this throws a null pointer exception because the non-generic tasks don't have the state. can we add a guard to prevent users from using this API for non-generic tasks?